src/ert/storage/migration/to4.py

```diff
@@ -14,14 +14,28 @@
 
 def migrate(path: Path) -> None:
     ert_config = local_storage_get_ert_config()
-    observations = ert_config.observations
+    observations = ert_config.observations.datasets
 
     for experiment in path.glob("experiments/*"):
         if observations:
             output_path = experiment / "observations"
             output_path.mkdir(parents=True, exist_ok=True)
-            for name, dataset in observations.items():
-                dataset.to_netcdf(output_path / f"{name}", engine="scipy")
+
+            for response_type, multi_obs_ds in observations.items():
```

Comment:
This should not be here?

Reply:
Not 100% on this but I think so yes. The storage is tied to the way the observations are stored in config, which is not really subject to storage migrations. The observations parsed from the ert config will now always be grouped by response type. But it still expects to get this dict of observations grouped by observation name.